js/components/meta-select-control.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies.
+ */
+const { __ } = wp.i18n;
+const { SelectControl } = wp.components;
+const { withSelect, withDispatch } = wp.data;
+const { compose } = wp.compose;
+
+const MetaSelectControl = compose(
+	withDispatch( function( dispatch, props ) {
+		return {
+			setMetaValue( value ) {
+				dispatch( 'core/editor' ).editPost( {
+					meta: { [ props.metaKey ]: value },
+				} );
+			},
+		};
+	} ),
+	withSelect( function( select, props ) {
+		const currentPostId = select( 'core/editor' ).getCurrentPostId();
+		const query = {
+			per_page: -1,
+			exclude: currentPostId,
+			metaKey: 'sub_menu',
+			metaValue: 0,
+		};
+		return {
+			metaValue: select( 'core/editor' ).getEditedPostAttribute( 'meta' )[ props.metaKey ],
+			adminPages: select( 'core' ).getEntityRecords( 'postType', 'admin-page', query ),
+		};
+	} )
+)( function( props ) {
+	const choices = [];
+	if ( props.adminPages ) {
+		choices.push( { value: 0, label: __( 'Select Parent Menu', 'hello-admin' ) } );
```

Comment:
You've used "Parent Menu" here (which is my preference) but it's under the heading "Menu Parent".

Reply:
Makes sense! not sure why I used sloppy English but its sorted now